Mathlib/AlgebraicGeometry/ProjectiveSpace.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Mathlib.AlgebraicGeometry.AffineSpace
+import Mathlib.AlgebraicGeometry.ProjectiveSpectrum.Proper
+import Mathlib.RingTheory.MvPolynomial.Homogeneous
+
+/-!
+# Projective space
+
+## Main definitions
+
+- `AlgebraicGeometry.ProjectiveSpace`: `ℙ(n; S)` is the projective `n`-space over `S`.
+-/
+
+/-
+TODO:
+- `AlgebraicGeometry.ProjectiveSpace.SpecIso`: `ℙ(n; Spec R) ≅ Proj R[n]`
```

Comment:
I would like to see some basic API in this PR for sanity check, in particular this thing should probably be in this PR.

Reply:
This is a bit hard at the moment, it's the main result in my personal branch and it takes 1000 lines. There's basically a lot of explicit maps between polynomial rings that are missing from the library, as the construction is by defining open covers on both sides, and using also isomorphisms on the intersections (i.e. pullbacks).

Does my personal branch work as the sanity check?

Reply:
Is this really the proof you want? I would have assumed that this should follow once you obtain the functor of points for `Proj Z[n]` (whose proof hopefully generalizes to `Proj R[n]` immediately)? Although that is indeed a lot of work.

Reply:
Or actually shouldn't you show that `Proj` is compatible with base change?

Reply:
These two alternative approaches are both out of scope for this PR which is to just get the basic definition in. Generalisations or not, I think we still ultimately want P(n; S) to exist.